File: go/problems/day7_test.go
package problems

import "testing"

var day7 Solvable[int64] = Day7{}

func TestDay7_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day7_1")
	result := day7.Solve(input)

	var expected int64 = 6440
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay7_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day7_2")
	result := day7.Solve(input)

	var expected int64 = 250957639
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day7.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day7.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 7", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day7_1");
    const result = solve(input);
    assert.equal(result, 6440);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day7_2");
    const result = solve(input);
    assert.equal(result, 250957639);
  });
});
